rpc/glfs-operations.c

```diff
@@ -119,6 +119,66 @@ glusterBlockCheckAvailableSpace(struct glfs *glfs,
 }
 
 
+int
+glusterBlockZeroFill(struct glfs_fd *tgfd, off_t offset, size_t size)
+{
+  struct iovec iov[4];
+  char *zerodata = NULL;
+  size_t chunksize = 4194304;  //4MiB
+  size_t len;
+  size_t rest;
+  int ret = -1;
+  int i;
+
+
+  LOG("gfapi", GB_LOG_INFO,
+      "zerofill is not supported for this volume type, slow zeroing will be used");
+
+  if (GB_ALLOC_N(zerodata, chunksize) < 0) {
+    LOG("gfapi", GB_LOG_ERROR, "Alloc failed");
+    goto out;
+  }
+
+  for(i = 0; i < 4; ++i) {
+    iov[i].iov_base = zerodata;
+    iov[i].iov_len = chunksize;
+  }
+
+  if (glfs_lseek(tgfd, offset, SEEK_SET) < 0) {
+    LOG("gfapi", GB_LOG_ERROR,
+        "glfs_lseek() failed: %s", strerror(errno));
+    goto out;
+  }
+
+  while (size > chunksize * 4) {
```

Comment:
@lxbsz and here it should be '>=' ?

Reply:
@pkalever  Yeah, that's right and it should be.